### DISCUSSION TITLE
Car not sleeping while this integration is enabled - default polling settings

### DISCUSSION THREAD

**Discussion by lellis2k** [alandtse/tesla#765] [Q&A]
My car was draining 9% in half a day so I changed my Tesla account password (to kick off other third party apps) and re-enabled only this integration. It still dropped by the same amount. 
I changed the password again, and didn't enable this integration and it now only drained 3% per day.

I posted this screenshot to FB and someone pointed out that the car must be awake this whole time for HA to have constantly changing data. How can I troubleshoot this to allow my car to sleep as much as it would without the integration? I'd like the data there when the car would be awake anyway, just don't want extra battery drain from it if possible. 2015 Model S 85.

![395425012_10160821848139174_3162632161537841073_n](https://github.com/alandtse/tesla/assets/3228976/4a1428f6-a7f8-4208-a031-bbc027a87cc4)

**Comment by skipishere** [comment]
We really would need some more information. Logs showing that the car is being woken up from the integration would indicate where the extension being at fault (if it is the case).

~Maybe a first step if you do think it is this, would be to deactivate the extension and see if you get similar batter decreases on its own.~ (Realised you mentioned that already) Remember, opening the Tesla app to view it will also potentially wake up the car as well.

Also if sentry mode is enabled there will be battery drain (this might be the cause, but again without more info we are purely guessing)

**Reply by lellis2k** [reply]
@skipishere thanks for the reply, 2015 so no sentry, decrease was far less when I disabled the integration and I'm definitely not waking the car via the app as you can see in the graph it's being kept awake all night while I'm asleep.

How can I retrieve the logs you're requesting as the next step then?

<img width="316" alt="image" src="https://github.com/alandtse/tesla/assets/3228976/9a52c12e-5521-4bd3-9052-d8fd75bc20a2">

**Reply by skipishere** [reply]
If you use this link https://my.home-assistant.io/redirect/integration/?domain=tesla_custom and then there should be an `Enable debug logging` button. Leaving that on for a while (maybe overnight?) and then once you disable it you should then have the option to download the log file.

**Reply by skipishere** [reply]
Also be interesting to see what the status history shows for the Online sensor. As ideally this extension won't wake the car up unless requested/action taken.
![image](https://github.com/alandtse/tesla/assets/26724777/8df01cac-51c4-4642-8454-574e57bc8819)

**Reply by lellis2k** [reply]
Enabled debug so will let you know on that one. 
Online sensor shows disconnected, for a period of time after I've re-enabled today but will have a better picture tomorrow.
<img width="393" alt="image" src="https://github.com/alandtse/tesla/assets/3228976/517fc6a7-f0ed-4b07-bfbb-716f849c7665">

**Comment by lellis2k** [comment]
oh and I think I remember this from before, it never sleeps per this sensor, does that work for yours? I don't know how I could monitor whether it's HA causing this because if I disable the add-on, what could I use instead to detect whether it's sleeping?
<img width="440" alt="image" src="https://github.com/alandtse/tesla/assets/3228976/f2cbfa16-d48f-4d5a-8149-c9ee2607dff0">

**Reply by skipishere** [reply]
Mine shows sleep most of the time. Unsure how you could check sleep status otherwise maybe someone else might know on that one (only had my Tesla since May)
![image](https://github.com/alandtse/tesla/assets/26724777/e7330113-297d-4532-8ece-2880f101b2ac)

**Reply by alandtse** [reply]
You can try increasing your polling interval from the default of 660 to see if there's some number that helps you. https://github.com/alandtse/tesla/wiki/Dynamic-Polling-Frequency#base-config.

**Comment by schind87** [comment]
FYI, I am experiencing the same issue and am now running debug for awhile to see and will post results here

**Comment by lellis2k** [comment]
[home-assistant_tesla_custom_2023-12-06T10-15-52.516Z.log](https://github.com/alandtse/tesla/files/13579792/home-assistant_tesla_custom_2023-12-06T10-15-52.516Z.log)
I re-enabled it for 24 hours, 5thDec23-6thDec23 10:15am (so scroll down about half way in the log) I did only lose 3% in that time, (which is good for my car), it _looks_ as though it didn't sleep according to the "asleep" sensor.
Does anyone understand the logging enough to see why?

Longest time it displays as last wake up is at 5:49am where it states "Last wake up 26619 ago." i.e. 7.4 hours. 
This lines up exactly with the "connected" sensor showing connected during this time frame:
![image](https://github.com/alandtse/tesla/assets/3228976/c6d44918-39b4-43b3-9b3c-c58dbe5fc191)
Log extract below:
```
2023-12-06 05:49:42.383 DEBUG (MainThread) [teslajsonpy.controller] Get vehicles. Force: False Time: 50 Interval 60
2023-12-06 05:49:42.384 DEBUG (MainThread) [teslajsonpy.controller] 59492: Skipping update with state offline. Polling: True. Last update: 3891 ago. Last parked: 70595 ago. Last wake up 26619 ago. 
```

During this time the numbers still appear to have been updated (thus, the car not sleeping?) as shown by the range changing values through the night:
![image](https://github.com/alandtse/tesla/assets/3228976/9f496990-49c1-4368-b29f-a640401ffa6c)

**Comment by ddaddy** [comment]
Mine is showing that it never sleeps too. 900 second intervals and sentry switched off.

**Comment by pacemaker82** [comment]
Wanted to add that I am now suffering this, after it being OK for a couple of days - the car constantly goes offline then online.

<img width="329" alt="Screenshot 2024-02-16 at 17 22 01" src="https://github.com/alandtse/tesla/assets/157808562/89293383-625c-4c37-9bdb-7114ba0ad9a7">
